.github/workflows/generateconfig.yml

```diff
@@ -0,0 +1,133 @@
+name: GenerateConfig
+
+on: workflow_dispatch
+
+jobs:
+  GenerateConfig:
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
+      contents: write
+    env: 
+      IAM_INSTANCE_PROFILE_ARN: ${{secrets.IAM_INSTANCE_PROFILE_ARN}} 
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Install xmllint
+      run: sudo apt-get update && sudo apt-get install libxml2-utils
+    - name: Configure AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v4
+      with: 
+        role-to-assume: ${{secrets.AMI_GENERATE_CONFIG_ROLE}}
+        aws-region: us-west-2
+    - name: Configure prereqs
+      run: |
+        git config --global user.name "GenerateConfig Action"
+        git config --global user.email "gcaction@github.com"
+        UpdateAl1="false"
+        UpdateAl2="false"
+        UpdateAl2023="false"
+    - name: Check AL1 Updates
+      run: |
+        source_ami_line_nums=(2 3 4 5)
```

Comment:
could we put all this bash in the repo `scripts/` folder rather than having it defined in in-line yaml?

Reply:
so that in yaml this just becomes something like `run: scripts/generate-al1-config` or `run: scripts/generate-config al1`

Reply:
Now that you mention it, I think this is a good idea. I will do so.

Reply:
done.